File: vmc/beginner/l2.4/Triangles.c
#include<stdio.h>

void print_left_triangle(int base)
{
    for (int i = 1; i < base+1; i++)
    {
        for (int j = 0; j < i; j++)
        {
            printf("*");
        }
        printf("\n");
    }
}

void print_left_triangle1(int base,char c)
{
    for (int i = 1; i < base+1; i++)
    {
        for (int j = 0; j < i; j++)
        {
            printf("%c",c);
        }
        printf("\n");
    }
}

void print_right_triangle(int base,char c)
{
    for (int i = 1; i < base+1; i++)
    {
        for (int k = 0; k < base-i; k++)
        {
            printf(" ");
        }
        
        for (int j = 0; j < i; j++)
        {
            printf("%c",c);
        }
        printf("\n");
    }
}


int main()
{
    print_left_triangle(20);
    print_left_triangle1(20,'%');
    print_right_triangle(20,'*'); // ' ' = char     " " = string
    return 1;
}

File: vm/beginner/l2.4/Triangles.py
def print_left_triangle(base):
    """prints an ASCII art right triangle of starts which grows 
        to 'base' stars.   For example print_left_triangle(3) prints
       *
       **
       ***
    """
    for i in range(base):
        print("*"*(i+1))

print_left_triangle(20)


def print_left_triangle1(base,char):
    for i in range(base):
        print( char*(i+1) )

print_left_triangle1(20,"$") #age "" nazari kar nmikone





def print_right_triangle(base):
    for i in range(20):
        print(" "*(base - i-1) , "*"*(i+1))

print_right_triangle(20)



